geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/ClearPRMessage.java

```diff
@@ -293,16 +225,21 @@ public ClearReplyMessage() {}
 
     private ClearReplyMessage(int processorId, boolean result, ReplyException ex) {
       super();
-      this.result = result;
       setProcessorId(processorId);
-      setException(ex);
+      if (ex != null) {
+        setException(ex);
+      } else {
+        setReturnValue(result);
+      }
     }
 
-    /** Send an ack */
+    /**
+     * Send an ack
+     */
     public static void send(InternalDistributedMember recipient, int processorId,
         ReplySender replySender,
         boolean result, ReplyException ex) {
-      Assert.assertTrue(recipient != null, "ClearReplyMessage NULL reply message");
+      Assert.assertTrue(recipient != null, "ClearReplyMessage NULL recipient.");
```

Comment:
There is an `Assert.assertNotNull()` that would be better here than `assertTrue()`.
Also, it might be clearer with a message of "ClearReplyMessage recipient was NULL."